### DISCUSSION TITLE
missing module named multiprocessing.get_context

### DISCUSSION THREAD

**Discussion by vaishnavireddipalli** [pyinstaller/pyinstaller#6584] [Help] [answered]
My program uses multiprocessing.get_context("spawn"), and the run gets stuck at this point of code, restarts again when I run with exe created by py installer. I am seeing "missing module named multiprocessing.get_context - imported by multiprocessing (top-level)" in the warning txt generated while running pyinstaller. 
I tried including hidden imports for multiprocessing, collect_all, add my conda env path still have the missing module. 

PyInstaller: 4.9
Python: 3.7.11 (conda)
On Linux
--onefile

I installed all required dependencies in conda environment, But I see 400 missing modules in warn txt.
Even running with onedir didn't help.
Please let me know if I am missing something.

**Comment by rokm** [comment, marked as answer]
Are you missing a `multiprocessing.freeze_support()` call at the very start of your entry-point? 

You'll need to provide a sample code that exhibits this problem; it's impossible to say anything without knowing what you're actually doing...

**Reply by vaishnavireddipalli** [reply]
Thank you @rokm. My application is running fine on Linux after adding multiprocessing.freeze_support().
But I still have a huge missing module (~400) list in the build folder warn.txt file, is that normal? (it's not affecting my application - I am not seeing any errors while running the pyinstaller executable)

**Reply by rokm** [reply]
> But I still have a huge missing module (~400) list in the build folder warn.txt file, is that normal?

It is normal - this happens because of conditional imports in various python packages that are (directly or indirectly) imported by your code. If any of those missing modules were problematic, you would either hit a missing-module error at run-time or, if the import was conditional, you would observe the corresponding functionality missing from the frozen application.

**Reply by vaishnavireddipalli** [reply]
Thank you @rokm for explaining.
